libmattermost.c

```diff
@@ -4101,352 +4314,319 @@ mm_get_chat_name(GHashTable *data)
-	mm_fetch_url(ma, url, NULL, mm_got_room_info, channel);
+	url = mm_build_url(ma,"/channels/%s/posts?page=%s&per_page=%s&since=%" G_GINT64_FORMAT "", channel->id, g_strdup_printf("%i",channel->page_history), g_strdup_printf("%i", MATTERMOST_HISTORY_PAGE_SIZE), since);
+	mm_fetch_url(ma, url, MATTERMOST_HTTP_GET, NULL, mm_got_history_of_room, channel);
 	g_free(url);
 }
 
-
 static void 
 mm_join_room(MattermostAccount *ma, MattermostChannel *channel)
-{
-	gchar *url;
-	url = mm_build_url(ma, "/api/" MATTERMOST_VERSION "/teams/%s/channels/%s/join", channel->team_id, channel->id);
-	mm_fetch_url(ma, url, "{}", mm_join_room_response, channel);
-	g_free(url);
+{	
+	mm_set_group_chat(ma, channel->team_id, channel->name, channel->id);
+	mm_get_users_of_room(ma, channel);
 }
 
+
 static void
 mm_join_chat(PurpleConnection *pc, GHashTable *chatdata)
 {
 	MattermostAccount *ma = purple_connection_get_protocol_data(pc);
 	const gchar *id = g_hash_table_lookup(chatdata, "id");
 	const gchar *name = g_hash_table_lookup(chatdata, "name");
 	const gchar *team_id = g_hash_table_lookup(chatdata, "team_id");
+	const gchar *type = g_hash_table_lookup(chatdata, "type");
+	const gchar *creator_id = g_hash_table_lookup(chatdata, "creator_id");
+
 	guint id_hash;
 	PurpleChatConversation *chatconv;
-	
+
 	if (id == NULL && name == NULL) {
 		//What do?
 		return;
 	}
-	
+
 	if (id == NULL) {
 		id = g_hash_table_lookup(ma->group_chats_rev, name);
 	}
 	//TODO use the api look up name info from the id
 	if (id == NULL) {
 		return;
 	}
-	
+
 	id_hash = g_str_hash(id);
 	chatconv = purple_conversations_find_chat(ma->pc, id_hash);
-	
+
 	if (chatconv != NULL && !purple_chat_conversation_has_left(chatconv)) {
 		purple_conversation_present(PURPLE_CONVERSATION(chatconv));
 		return;
 	}
-	
-	mm_set_group_chat(ma, team_id, name, id); 
 
-	chatconv = purple_serv_got_joined_chat(pc, id_hash, name);
+	const gchar *alias = g_hash_table_lookup(ma->aliases,id);
+
+	chatconv = purple_serv_got_joined_chat(pc, id_hash, alias);//ALIAS ?
+//	purple_serv_got_chat_in(ma->pc, g_str_hash(id), "none", PURPLE_MESSAGE_SYSTEM, "no message", 0);
+
 	purple_conversation_set_data(PURPLE_CONVERSATION(chatconv), "id", g_strdup(id));
 	purple_conversation_set_data(PURPLE_CONVERSATION(chatconv), "team_id", g_strdup(team_id));
 	purple_conversation_set_data(PURPLE_CONVERSATION(chatconv), "name", g_strdup(name));
+	purple_conversation_set_data(PURPLE_CONVERSATION(chatconv), "type", g_strdup(type));
+	purple_conversation_set_data(PURPLE_CONVERSATION(chatconv), "creator_id", g_strdup(creator_id));
 	purple_conversation_present(PURPLE_CONVERSATION(chatconv));
-	
+
 	MattermostChannel *channel = g_new0(MattermostChannel,1);
 	channel->name = g_strdup(name);
 	channel->id = g_strdup(id);
 	channel->team_id = g_strdup(team_id);
+	channel->type = g_strdup(type);
 
-	mm_join_room(ma, channel);
+	mm_get_channel_by_id(ma,team_id,id);
+//	mm_join_room(ma, channel);

```

Comment:
Not using the channel object causes a memleak, maybe remove the creation of the MattermostChannel completely?

Reply:
This is the only major memory leak that I can find from a static analyzer